epyqlib/devicetreeview.py

```diff
@@ -221,6 +230,28 @@ def add_device(self, bus, device=None):
 
         return device
 
+    def change_device_number(self, node):
+        pcan_bus = can.interface.Bus(bustype=node.interface, channel=node.channel)
+        current_device_number = pcan_bus.get_device_number()
+        new_device_number, ok = QInputDialog.getInt(
+            None,
+            *(("Device ID",) * 2),
+            current_device_number,
+            0,
+            255,
+        )
+
+        if ok:
+            if pcan_bus.set_device_number(new_device_number):
+                node.set_device_number(new_device_number)
+            else:
+                pcan_bus.shutdown()
+                raise Exception(
+                    f"Unable to change device number from '{current_device_number}' to '{new_device_number}'."
+                )
+
+        pcan_bus.shutdown()
```

Comment:
Probably ought to just `try`/`finally` the whole function body to make sure the bus gets shutdown in all cases.  Unless maybe it can be a context manager?

Reply:
Yes. I agree. I didn't like how I did this one. I did try/except/finally because a context manager didn't work.